packages/unified-server/src/landing/main.ts

```diff
@@ -221,6 +223,26 @@ async function init() {
         sharedFlowDialog.close();
       });
     }
+
+    const handleSecondaryVideoIframeClick = () => {
+      setTimeout(() => {
+        if (document.activeElement === secondaryVideo) {
+          if (secondaryVideoContainer) {
+            const secondaryVideoCover = secondaryVideoContainer.querySelector(".dimmed-cover");
+            secondaryVideoCover?.remove();
+          }
+          window.removeEventListener('blur', handleSecondaryVideoIframeClick);
+        }
+      }, 0);
+    }
+
+    // gain window focus so that blur even will fire without any user interaction
+    if (document.hasFocus() === false) {
+      window.focus();
+      document.body?.focus?.();
+    }
+
+    window.addEventListener('blur', handleSecondaryVideoIframeClick);
```

Comment:
What is the approach taken here? We seem to be adding a blur handler to the window which triggers some DOM updates?